CompetitionTemplate/src/competition/subsystems/autonomous/DriveForDistanceCommand.java

```diff
@@ -0,0 +1,60 @@
+package competition.subsystems.autonomous;
+
+import com.google.inject.Inject;
+
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.drive.PoseSubsystem;
+import xbot.common.command.BaseCommand;
+import xbot.common.math.PIDManager;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class DriveForDistanceCommand extends BaseCommand{
+    DriveSubsystem driveSubsystem; 
+    PoseSubsystem poseSubsystem;
+    
+    DoubleProperty distancePerTick;
+    DoubleProperty targetRange;
+    
+    PIDManager distanceDrivePid;
+    
+    public double targetDistance;
+    double targetEncoderDistance;
+    double currentDistance;
+    
+    @Inject
+    public DriveForDistanceCommand(PoseSubsystem poseSubsystem, DriveSubsystem driveSubsystem, XPropertyManager propManager){
+        this.driveSubsystem = driveSubsystem;
+        this.poseSubsystem = poseSubsystem;
+        
+        distancePerTick = propManager.createPersistentProperty("the distance per tick on encoder", 10.0);
+        distanceDrivePid = new PIDManager("DistanceDrivePID", propManager, 1, 0, 0);
+        targetRange = propManager.createPersistentProperty("DistanceTargetRange", 3.0);
+    }
+    
+    public void setTargetDistance(double distance) {
+        targetDistance = distance;
```

Comment:
are we expecting this distance to be relative to the robots current position or absolute? I'm guessing relative but in that case @JohnGilb I think we'll need to record our current distance in initialize and look at those deltas